### PR TITLE
[consensus] check rpc epoch in epoch_manager

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

This PR implements epoch check before processing any incoming RPC request.

### CODE REVIEW

**Comment by ibalajiarun** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #12018** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/12018?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/12018?utm_source=stack-comment).

File: consensus/src/network.rs [hunk 1]
```diff
@@ -133,6 +133,17 @@ pub enum IncomingRpcRequest {
     RandGenRequest(IncomingRandGenRequest),
 }
 
+impl IncomingRpcRequest {
+    pub fn epoch(&self) -> Option<u64> {
+        match self {
+            IncomingRpcRequest::BatchRetrieval(req) => Some(req.req.epoch()),
+            IncomingRpcRequest::DAGRequest(req) => Some(req.req.epoch()),
+            IncomingRpcRequest::RandGenRequest(req) => Some(req.req.epoch()),
+            IncomingRpcRequest::BlockRetrieval(_) | IncomingRpcRequest::CommitRequest(_) => None,
```
Comment (zekun000) [inline consensus/src/network.rs:142]: we should be able to parse epoch numbers from CommitRequest that's what actually log errors too

Comment (ibalajiarun) [reply consensus/src/network.rs]: Updated the PR.

**Review by zekun000 — APPROVED** [review]

File: consensus/src/epoch_manager.rs [hunk 2]
```diff
@@ -1395,6 +1395,19 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
...
+
+        match request.epoch() {
+            Some(epoch) if epoch != self.epoch() => {
+                monitor!(
+                    "process_different_epoch_rpc_request",
+                    self.process_different_epoch(epoch, peer_id)
+                )?;
+                return Ok(());
+            },
+            // TODO: differentiate between None and same epoch.
```
Comment (bchocho) [inline consensus/src/epoch_manager.rs:1407]: Should None at least return early?

Comment (ibalajiarun) [reply consensus/src/epoch_manager.rs:1407]: No, BlockRetrieval doesn't have an epoch number for instance.

**Review by bchocho — APPROVED** [review]

**Pull request merged**